packages/backend/server/src/plugins/copilot/providers/types.ts

```diff
@@ -107,81 +102,45 @@ export const CopilotImageOptionsSchema = CopilotProviderOptionsSchema.merge(
 
 export type CopilotImageOptions = z.infer<typeof CopilotImageOptionsSchema>;
 
-export interface CopilotTextToTextProvider extends CopilotProvider {
-  generateText(
-    messages: PromptMessage[],
-    model: string,
-    options?: CopilotChatOptions
-  ): Promise<string>;
-  generateTextStream(
-    messages: PromptMessage[],
-    model: string,
-    options?: CopilotChatOptions
-  ): AsyncIterable<string>;
-}
+export const CopilotEmbeddingOptionsSchema =
+  CopilotProviderOptionsSchema.extend({
+    dimensions: z.number(),
+  }).optional();
 
-export interface CopilotTextToEmbeddingProvider extends CopilotProvider {
-  generateEmbedding(
-    messages: string[] | string,
-    model: string,
-    options?: CopilotEmbeddingOptions
-  ): Promise<number[][]>;
+export type CopilotEmbeddingOptions = z.infer<
+  typeof CopilotEmbeddingOptionsSchema
+>;
+
+export enum ModelInputType {
+  Text = 'text',
+  Image = 'image',
+  Audio = 'audio',
 }
 
-export interface CopilotTextToImageProvider extends CopilotProvider {
-  generateImages(
-    messages: PromptMessage[],
-    model: string,
-    options?: CopilotImageOptions
-  ): Promise<Array<string>>;
-  generateImagesStream(
-    messages: PromptMessage[],
-    model: string,
-    options?: CopilotImageOptions
-  ): AsyncIterable<string>;
+export enum ModelOutputType {
+  Text = 'text',
+  Reasoning = 'reasoning',
```

Comment:
Is it necessary to distinguish the reasoning type? I feel that it currently belongs to the text type.

Reply:
used to distinguish models that do not support reasoning